Navbar#cl2 6698

sergei-kucher opened the pull request:
## Checklist

- [x] Added entry to changelog
<details>
<summary>More info</summary>
Add a concise line to the 'Next release' section of the changelog (docs/README.md) so people other than developers can understand what has changed where. E.g. 'Added an error message to the project name field of the project edit form (Admin > Projects > Edit)'.
</details>

- [ ] Tests
<details>
<summary>More info</summary>

### Unit tests

Did you add relevant unit tests?

### E2E tests

Sometimes it can be more efficient to update E2E tests after CI has run them. If you know which ones to update, go ahead! E2E template cl2-back:

```bash
docker compose run --rm web bin/rails cl2_back:create_tenant[localhost,e2etests_template]
```

</details>

- [ ] Prepared branch for code review
<details>
<summary>More info</summary>
Reviewed code to reduce unnecessary back and forth (removal of console.log, comments, ...)? Added comments to clarify code, emphasize what to pay attention to, etc.?
</details>

## Links

- [Jira ticket](**put URL here**)
- [citizenlab-ee PR](**put URL here** or remove)
- [Specs](**put URL here** or remove)
- [Epic Deployment](**put URL here** or remove)

## What changes are in this PR?

Add a concise summary here of what happened, so the reviewer has some background.

## How urgent is a code review?

Let the reviewer(s) know how urgent the code review is, so they can prioritize their work accordingly. Be specific (e.g. by Wednesday, end of the day/this week/... is better than 'urgent' or 'very urgent'). Optionally provide a word of explanation on your deadline.


- [ ] Change back the postgres image in docker-compose
- [x] Don't merge to master. Create a branch for customizable navbar

sergei-kucher committed:
Customisable navigation bar

sergei-kucher committed:
done

sergei-kucher committed:
specs

sergei-kucher committed:
-

sergei-kucher commented:
@kogre Something went wrong with the previous pull-request (https://github.com/CitizenLabDotCo/citizenlab/pull/716). It didn't show diffs conrrectly. I had to close it.

sergei-kucher commented on back/app/models/navbar_item.rb:
```diff
@@ -0,0 +1,106 @@
+    if ordering > (self.class.where(visible: visible).maximum_ordering + 1)
+      errors.add(:ordering, "cannot be aside from others")
+    end
+  end
```
From @kogre
> Have bad experience with such validations (that are not just about the current record, but depend on other records). These will work as desired in most scenarios, but will break when e.g. you create some items in bulk, delete an item in-between, etc.

> In the rest of the code, we're using acts_as_list gem for it. It works pretty well, and by itself deals with list-related validations. Not sure whether you saw that, but if you did, any reason not to adopt it here?

sergei-kucher replied:
I looked at `acts_as_list` and it looks like it fits here. It's not clear for me yet how validations work in this gem.

I'm trying refactor the PR to use this gem. Hope it'll work out.

sergei-kucher replied:
@kogre

sergei-kucher replied:
By the way, I don't really like rails validations. I think they are messy and don't allow to write clean code. I made use of them for navbar only because the project uses them everywhere.

I used to use `dry-validations`. They allow to separate validations from models and make everything much cleaner.

If you don't mind, I would like to use `dry-validations` for the next epic.

kogre replied:
You can maybe give a small demo in the next BE CDM next Thursday on dry-validations, to see if we find it worthwhile to try? Feel free to add it to the agenda.

sergei-kucher commented on back/app/models/navbar_item.rb:
```diff
@@ -0,0 +1,106 @@
+  def cannot_destroy_reserved
+    if ordering <= LAST_RESERVED_ORDERING
+      errors.add :ordering,
+        "Cannot destroy a reserved navbar item. Ordering (#{ordering}) should be > #{LAST_RESERVED_ORDERING}."
```
From @kogre 
> Since our error in the end reach the FE, we work with codes instead of text. Or both, but at least we need the code, so the FE can interpret it and show the right (multilingual) copy.

> So this should become something like `errors.add(:ordering, :cannot_destroy_reserved, message: "Cannot destr... ")`

sergei-kucher replied:
Ok. I'll fix it.

sergei-kucher commented on back/app/models/navbar_item.rb:
```diff
@@ -0,0 +1,106 @@
+    return if visible_items_count < MAX_VISIBLE_ITEMS
+
+    errors.add :visible, "Cannot make the item visible when the list of visible items is full (max: #{MAX_VISIBLE_ITEMS})"
+  end
```
From @kogre 
> Same here, validations depending on other records have a tendency to be problematic. They are prone to race conditions like bulk updates, or cached queries. If you're confident it works in all cases performed by the API, I can live with it.

sergei-kucher replied:
Since everything what controllers do is wrapped in transactions, I don't really see any serious issues with it.

Only potential issue I see in this case is that if two identical requests are sent at the same time the navbar items list could contain more than `MAX_VISIBLE_ITEMS`.

But it's highly unlikely and can be fixed by removing the excess item from the list.

Anyway I don't really see any alternatives to this approach.

sergei-kucher replied:
@kogre

sergei-kucher commented on back/app/models/page.rb:
```diff
@@ -28,6 +35,55 @@ def published?
+    return unless slug_changed?
+
+    if navbar_item.nil?
+      errors.add :slug, <<~TXT.split("\n").join(" ")
```
From @kogre 
> Using codes for errors in this file as well.

sergei-kucher replied:
Ok. I'll fix it.

kogre commented on docker-compose.yml:
```diff
@@ -2,7 +2,8 @@ version: "3.8"
 services:
   postgres:
-    image: "postgis/postgis:12-3.1"
+    # TODO change back before merging
+    image: "skucher/postgis:12-3.1"
```
Is this still needed for the M1?

kogre reviewed (approved):
Nice work @sergey-kucher 

I haven't been reading every line in detail, but the extensive test coverage and the general clarity of the code gives confidence.

Apart from the points I remarked (returning error codes and checking whether acts_as_list can't simplify the most complicated parts of this PR) I didn't spot any other obvious things to remark.

adessy committed:
IN-548 Filter inputs by keywords (simplified version -- without lemmas)

IN-548 #time 0w 0d 0h 1m Filter inputs by keywords (simplified version -- without lemmas)

adessy committed:
IN-548 Permit `keywords` query parameter

adessy committed:
IN-548 Add 'keywords' parameter to the documentation

brentguf committed:
Revert "Attempt to fix uploading issues"

This reverts commit a2f236c11982467b8d4d8d2d833fb82c86727cde.

adessy committed:
IN-548 Delegate methods to the nlp networks

IN-548 #time 0w 0d 0h 38m Delegate methods to the nlp networks

adessy committed:
IN-548 Skip filter when the keyword parameter is not present (covers the empty array case)

IN-548 #time 0w 0d 0h 1m Skip filter when the keyword parameter is not present (covers the empty array case)

adessy committed:
IN-548 Add acceptance test from the keyword filter (simplified -- without lemmas)

IN-548 #time 0w 0d 0h 1m Add acceptance test from the keyword filter (simplified -- without lemmas)

IvaKop committed:
Update snapshot

IvaKop committed:
Merge pull request #785 from CitizenLabDotCo/dependabot/npm_and_yarn/front/recharts-2.1.4

Bump recharts from 2.0.9 to 2.1.4 in /front

kogre committed:
New translations en.json (German)

brentguf committed:
PlatformFooter to functional components

adessy committed:
IN-548 Test filtering by keyword

IN-548 #time 0w 0d 0h 8m Test filtering by keyword

sergei-kucher committed:
f

brentguf committed:
Up

brentguf committed:
Merge pull request #788 from CitizenLabDotCo/crowdin_master

New Crowdin updates

brentguf committed:
Add Next release section

brentguf committed:
Merge pull request #797 from CitizenLabDotCo/changelog/release-2021-09-29

changelog/release-2021-09-29

adessy committed:
IN-548 Rework InputsFinder interface (merging options & params into a single parameter) + a few tests

IN-548 #time 0w 0d 0h 57m Rework InputsFinder interface (merging options & params into a single parameter) + a few tests

brentguf committed:
Extract ShortFeedback from PlatformFooter

kogre committed:
New translations en.json (Portuguese, Brazilian)

brentguf committed:
Fix closing of modal on cancel

brentguf committed:
Don't show info/FAQ in footer

brentguf committed:
Remove PagesFooterNavigation from pages

brentguf committed:
Merge pull request #798 from CitizenLabDotCo/crowdin_master

New Crowdin updates

Commit:
Translations updated by CI (extract-intl)

brentguf committed:
Remove PagesFooterNavigation, pt2

brentguf committed:
Merge branch 'CL2-6674/remove-items-footer-custom-pages' of https://github.com/CitizenLabDotCo/citizenlab into CL2-6674/remove-items-footer-custom-pages

brentguf committed:
Merge branch 'master' of https://github.com/CitizenLabDotCo/citizenlab into CL2-6674/remove-items-footer-custom-pages

brentguf committed:
Changelog

adessy committed:
IN-548 Disable pagination for xlsx export of inputs + test

IN-548 #time 0w 0d 2h 15m Disable pagination for xlsx export of inputs + test

IvaKop committed:
Add keywords to url

adessy committed:
IN-548 Disable pagination by default in InputsFinder

IN-548 #time 0w 0d 0h 1m Disable pagination by default in InputsFinder

adessy committed:
IN-548 Use the node.name instead of node.id (even if they should be equal in that case)

IN-548 #time 0w 0d 0h 18m Use the node.name instead of node.id (even if they should be equal in that case)

adessy committed:
IN-548 Fix typo [skip ci]

IN-548 #comment IN-548 #time 0w 0d 0h 1m Fix typo

brentguf committed:
pages.ts types

brentguf committed:
PageEditor general pageSlug prop type

brentguf committed:
Build errors

brentguf committed:
Build error

brentguf committed:
Comment out short feedback form

adessy committed:
IN-548 Fix tests

IN-548 #time 0w 0d 1h 39m Fix tests

adessy committed:
IN-548 Sort query terms to keep the same ranking (inputs order) for a given set of keywords (whatever their order)

IN-548 #time 0w 0d 0h 6m Sort query terms to keep the same ranking (inputs order) for a given set of keywords (whatever their order)

IvaKop committed:
Add request

IvaKop committed:
Merge branch 'IN-548/keyword-filter-for-inputs' into IN-482/keyword-filter-for-inputs

kogre committed:
New translations en.json (Danish)

kogre committed:
New translations en.json (French, Belgium)

kogre committed:
New translations en.json (Spanish, Chile)

IvaKop committed:
Add export functionality

IvaKop committed:
Add intl

IvaKop committed:
Remove options

Commit:
Translations updated by CI (extract-intl)

IvaKop committed:
Remove comment

IvaKop committed:
Merge branch 'IN-484/export-viz' of github.com:CitizenLabDotCo/citizenlab into IN-484/export-viz

IvaKop committed:
Fix placement

sebastienhoorens committed:
added spec to check if all blocked words lists are available

sebastienhoorens committed:
ensuring blocked words work and spec for availability of lists

sebastienhoorens committed:
initial blocked lists for LU and IT

sebastienhoorens committed:
italian blocked words v1

sebastienhoorens committed:
LU -> LB

sebastienhoorens committed:
updated changelog

SaraBCL committed:
Reverts mistaken Typeform config overwrite

SaraBCL committed:
Merge branch 'master' of github.com:CitizenLabDotCo/citizenlab into IN-534/fix-typeform-exports

adessy committed:
IN-562 Implement categories[] filter

IN-562 #time 0w 0d 7h 46m Implement categories[] filter

adessy committed:
IN-562 Permit categories[] query parameter

adessy committed:
IN-562 Check for the existence of categories

IN-562 #time 0w 0d 0h 3m Check for the existence of categories

sebastienhoorens committed:
empty last line

sebastienhoorens committed:
empty last line

adessy committed:
IN-562 Merge the category filter into the categories filter (eventually the category filter will be removed)

IN-562 #time 0w 0d 0h 26m Merge the category filter into the categories filter (eventually the category filter will be removed)

brentguf committed:
Remove page_links

adessy committed:
IN-562 Add tests for the categories filter

IN-562 #time 0w 0d 0h 56m Add tests for the categories filter

adessy committed:
IN-562 Move unknown category to context

adessy committed:
Merge pull request #804 from CitizenLabDotCo/IN-562/filter-inputs-multiple-categories

brentguf committed:
Merge pull request #790 from CitizenLabDotCo/epic-review/fix-page-forms

Epic review/fix page forms

brentguf committed:
Merge pull request #800 from CitizenLabDotCo/crowdin_master

New Crowdin updates

brentguf committed:
Revert getInitialValues function

brentguf committed:
Build error

brentguf committed:
Merge branch 'master' of https://github.com/CitizenLabDotCo/citizenlab into CL2-6715/fix-page-forms

IvaKop committed:
Fix tests

adessy committed:
IN-563 Extract input-filter parameters that will be used to pre-populate the category

IvaKop committed:
Add categories to hook

IvaKop committed:
Add categories to url

SaraBCL committed:
Merge pull request #791 from CitizenLabDotCo/IN-534/fix-typeform-exports

IN-535/fix-typeform-exports

adessy committed:
IN-563 Pre-populate newly created categories if inputs parameters are present

IN-563 #time 0w 0d 1h 34m Pre-populate newly created categories if inputs parameters are present

adessy committed:
IN-563 Rework Insights::WebApi::V1::CategoriesController#create

IN-563 #time 0w 0d 0h 12m Rework Insights::WebApi::V1::CategoriesController#create

brentguf committed:
Changelog

IvaKop committed:
Display selected categories list

brentguf committed:
Changelog correction

adessy committed:
IN-563 Document new parameters for POST .../categories

IN-563 #time 0w 0d 0h 19m Document new parameters for POST .../categories

kogre committed:
New translations en.json (English, United Kingdom)

kogre committed:
New translations en.json (Dutch, Belgium)

kogre committed:
New translations en.json (Dutch)

brentguf committed:
Merge pull request #808 from CitizenLabDotCo/crowdin_master

New Crowdin updates

brentguf committed:
Merge pull request #718 from CitizenLabDotCo/CL2-6715/fix-page-forms

CL2-6715/fix page forms

IvaKop committed:
Fix empty state

kogre committed:
Update source file en.json

brentguf committed:
Merge branch 'master' of https://github.com/CitizenLabDotCo/citizenlab into CL2-6674/remove-items-footer-custom-pages

adessy committed:
IN-563 Test the creation of pre-populated categories

IN-563 #time 0w 0d 2h 9m Test the creation of pre-populated categories

adessy committed:
IN-563 Fix: remove 'processed' parameter from the documentation [skip ci]

IN-563 #time 0w 0d 0h 7m Fix: remove 'processed' parameter from the documentation [skip ci]

adessy committed:
IN-563 Fix tests

IN-563 #time 0w 0d 0h 17m Fix tests

IvaKop committed:
Add tracks

brentguf committed:
Revert "CL2-6715/fix page forms"

brentguf committed:
Merge pull request #813 from CitizenLabDotCo/revert-718-CL2-6715/fix-page-forms

Revert "CL2-6715/fix page forms"

kogre committed:
New translations en.json (English)

kogre committed:
New translations en.json (German)

dependabot[bot] committed:
Bump @types/leaflet.markercluster from 1.4.4 to 1.4.5 in /front

Bumps [@types/leaflet.markercluster](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/leaflet.markercluster) from 1.4.4 to 1.4.5.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/leaflet.markercluster)

---
updated-dependencies:
- dependency-name: "@types/leaflet.markercluster"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump d3 from 7.0.3 to 7.1.1 in /front

Bumps [d3](https://github.com/d3/d3) from 7.0.3 to 7.1.1.
- [Release notes](https://github.com/d3/d3/releases)
- [Changelog](https://github.com/d3/d3/blob/main/CHANGES.md)
- [Commits](https://github.com/d3/d3/compare/v7.0.3...v7.1.1)

---
updated-dependencies:
- dependency-name: d3
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @types/react from 17.0.22 to 17.0.27 in /front

Bumps [@types/react](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react) from 17.0.22 to 17.0.27.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react)

---
updated-dependencies:
- dependency-name: "@types/react"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump graphql from 15.5.1 to 15.6.0 in /front

Bumps [graphql](https://github.com/graphql/graphql-js) from 15.5.1 to 15.6.0.
- [Release notes](https://github.com/graphql/graphql-js/releases)
- [Commits](https://github.com/graphql/graphql-js/compare/v15.5.1...v15.6.0)

---
updated-dependencies:
- dependency-name: graphql
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump yargs from 15.4.1 to 17.2.1 in /front

Bumps [yargs](https://github.com/yargs/yargs) from 15.4.1 to 17.2.1.
- [Release notes](https://github.com/yargs/yargs/releases)
- [Changelog](https://github.com/yargs/yargs/blob/main/CHANGELOG.md)
- [Commits](https://github.com/yargs/yargs/compare/v15.4.1...v17.2.1)

---
updated-dependencies:
- dependency-name: yargs
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

luucvanderzee committed:
Merge pull request #814 from CitizenLabDotCo/dependabot/npm_and_yarn/front/types/leaflet.markercluster-1.4.5

Bump @types/leaflet.markercluster from 1.4.4 to 1.4.5 in /front

luucvanderzee committed:
Replaced d3 by d3-force

luucvanderzee committed:
Replaced d3 by d3-force

luucvanderzee committed:
Merge pull request #822 from CitizenLabDotCo/dependabot/npm_and_yarn/front/yargs-17.2.1

Bump yargs from 15.4.1 to 17.2.1 in /front

IvaKop committed:
Remove tsignore

luucvanderzee committed:
Merge pull request #816 from CitizenLabDotCo/dependabot/npm_and_yarn/front/d3-7.1.1

Replace d3 by d3-hierarchy

sebastienhoorens committed:
disable throttling through env variable

sebastienhoorens committed:
updated changelog

IvaKop committed:
Make comment more explicit

IvaKop committed:
Merge branch 'master' into IN-484/export-viz

IvaKop committed:
Merge branch 'master' into IN-484/export-viz

adessy committed:
Merge branch 'master' into IN-548/keyword-filter-for-inputs

IN-548 #comment # Conflicts:
#	back/engines/commercial/insights/app/controllers/insights/web_api/v1/inputs_controller.rb
#	back/engines/commercial/insights/app/finders/insights/inputs_finder.rb
#	back/engines/commercial/insights/spec/finders/insights/inputs_finder_spec.rb

IN-548 #time 0w 2d 3h 9m Merge branch 'master' into IN-548/keyword-filter-for-inputs

IvaKop committed:
Merge pull request #801 from CitizenLabDotCo/IN-484/export-viz

IN-484 Export visualization

kogre committed:
New translations en.json (Romanian)

kogre committed:
New translations en.json (English)

kogre committed:
New translations en.json (French, Belgium)

kogre committed:
New translations en.json (Dutch, Belgium)

kogre committed:
New translations en.json (Serbian (Latin))

kogre committed:
New translations en.json (Luxembourgish)

kogre committed:
New translations en.json (English, United Kingdom)

kogre committed:
New translations en.json (English, Canada)

kogre committed:
New translations en.json (Maori)

kogre committed:
New translations en.json (Spanish, Chile)

kogre committed:
New translations en.json (Portuguese, Brazilian)

kogre committed:
New translations en.json (Swedish)

kogre committed:
New translations en.json (French)

kogre committed:
New translations en.json (Polish)

kogre committed:
New translations en.json (Norwegian)

kogre committed:
New translations en.json (Dutch)

kogre committed:
New translations en.json (Italian)

kogre committed:
New translations en.json (Hungarian)

kogre committed:
New translations en.json (German)

kogre committed:
New translations en.json (Danish)

kogre committed:
New translations en.json (Arabic)

kogre committed:
New translations en.json (Spanish)

kogre committed:
New translations en.json (Greenlandic)

kogre committed:
New translations en.json (German)

adessy committed:
Merge pull request #794 from CitizenLabDotCo/IN-548/keyword-filter-for-inputs

adessy committed:
Merge pull request #806 from CitizenLabDotCo/IN-563/create-category-from-filter

kogre committed:
New translations en.json (Spanish)

IvaKop committed:
Merge master

sebastienhoorens committed:
try disabling rack attack in tests later

sebastienhoorens committed:
try disabling rack attack in tests later

IvaKop committed:
Merge pull request #799 from CitizenLabDotCo/IN-482/keyword-filter-for-inputs

IN-482 Keyword filter for inputs

IvaKop committed:
Merge master

kogre committed:
New translations en.json (German)

sebastienhoorens committed:
Merge pull request #825 from CitizenLabDotCo/CL2-6771-disable-rack-attack-env-var

CL2-6771 Disable throttling through env variable

sebastienhoorens committed:
Merge branch 'master' into CL2-6767-forgotten-blocked-words-lists

sebastienhoorens committed:
Merge pull request #802 from CitizenLabDotCo/CL2-6767-forgotten-blocked-words-lists

CL2-6767 Forgotten blocked words lists

kogre committed:
New translations en.json (Spanish, Chile)

adessy committed:
IN-570 Fix inputs sorting according the approval status (#827)

adessy committed:
[HOTFIX] Fix: undefined method `one?' for 0:Integer

kogre committed:
New translations en.json (Italian)

kogre committed:
New translations en.json (Danish)

brentguf committed:
Merge branch 'master' of https://github.com/CitizenLabDotCo/citizenlab into CL2-6674/remove-items-footer-custom-pages

brentguf committed:
Merge pull request #810 from CitizenLabDotCo/crowdin_master

New Crowdin updates

brentguf committed:
Merge pull request #795 from CitizenLabDotCo/CL2-6674/remove-items-footer-custom-pages

CL2-6674/Footer and custom page links

brentguf committed:
Up

brentguf committed:
Remove about and order of items

brentguf committed:
Merge pull request #830 from CitizenLabDotCo/CL2-6674/remove-items-footer-custom-pages

Remove about and order of items

kogre committed:
Update source file en.json

kogre committed:
New translations en.json (Romanian)

kogre committed:
New translations en.json (French, Belgium)

kogre committed:
New translations en.json (Dutch, Belgium)

kogre committed:
New translations en.json (Serbian (Latin))

kogre committed:
New translations en.json (Luxembourgish)

kogre committed:
New translations en.json (English, United Kingdom)

kogre committed:
New translations en.json (English, Canada)

kogre committed:
New translations en.json (Maori)

kogre committed:
New translations en.json (Spanish, Chile)

kogre committed:
New translations en.json (Portuguese, Brazilian)

kogre committed:
New translations en.json (Swedish)

kogre committed:
New translations en.json (French)

kogre committed:
New translations en.json (Polish)

kogre committed:
New translations en.json (Norwegian)

kogre committed:
New translations en.json (Dutch)

kogre committed:
New translations en.json (Italian)

kogre committed:
New translations en.json (Hungarian)

kogre committed:
New translations en.json (German)

kogre committed:
New translations en.json (Danish)

kogre committed:
New translations en.json (Arabic)

kogre committed:
New translations en.json (Spanish)

kogre committed:
New translations en.json (Greenlandic)

brentguf committed:
Merge pull request #829 from CitizenLabDotCo/changelog/release-2021-10-05

changelog/release-2021-10-05

alexander-cit committed:
Change front API_HOST to localhost

Default `.env-front.example` doesn't work on clean setup.
`API_HOST=cl-back-web` should be `API_HOST=localhost` instead,
because docker-compose doesn't create `cl-back-web` hostname
globally (on host machine), but only for docker instances
https://docs.docker.com/compose/networking/

It stopped working after this commit
https://github.com/CitizenLabDotCo/citizenlab/commit/2da75b8,
where we removed front service from docker-compose.yml.

IvaKop committed:
Merge pull request #807 from CitizenLabDotCo/IN-480/categories-filter-for-inputs

IN-480 Categories filter for inputs

alexander-cit committed:
Merge pull request #832 from CitizenLabDotCo/front-API_HOST-env-fixed

Change front API_HOST to localhost

IvaKop committed:
Merge master

IvaKop committed:
Add filters track

luucvanderzee committed:
Upgrade apollo too

luucvanderzee committed:
Merge pull request #820 from CitizenLabDotCo/dependabot/npm_and_yarn/front/graphql-15.6.0

Bump graphql from 15.5.1 to 15.6.0 in /front

IvaKop committed:
Move inital render condition

luucvanderzee committed:
Merge pull request #819 from CitizenLabDotCo/dependabot/npm_and_yarn/front/types/react-17.0.27

Bump @types/react from 17.0.22 to 17.0.27 in /front

IvaKop committed:
Merge pull request #812 from CitizenLabDotCo/IN-564/view-tracking

Add tracking for Insights View screen

luucvanderzee committed:
Upgraded @sentry/integrations too

luucvanderzee committed:
Merge branch 'master' into dependabot/npm_and_yarn/front/sentry/browser-6.13.2

luucvanderzee committed:
Merge pull request #784 from CitizenLabDotCo/dependabot/npm_and_yarn/front/sentry/browser-6.13.2

Bump @sentry/browser from 5.15.5 to 6.13.2 in /front

kogre committed:
New translations en.json (Romanian)

kogre committed:
New translations en.json (English)

kogre committed:
New translations en.json (French, Belgium)

kogre committed:
New translations en.json (Dutch, Belgium)

kogre committed:
New translations en.json (Serbian (Latin))

kogre committed:
New translations en.json (English, United Kingdom)

kogre committed:
New translations en.json (English, Canada)

kogre committed:
New translations en.json (Spanish, Chile)

kogre committed:
New translations en.json (Portuguese, Brazilian)

kogre committed:
New translations en.json (French)

kogre committed:
New translations en.json (Polish)

kogre committed:
New translations en.json (Norwegian)

kogre committed:
New translations en.json (Dutch)

kogre committed:
New translations en.json (German)

kogre committed:
New translations en.json (Danish)

kogre committed:
New translations en.json (Arabic)

kogre committed:
New translations en.json (Spanish)

kogre committed:
New translations en.json (Greenlandic)

kogre committed:
Update source file en.json

kogre committed:
New translations en.json (English)

kogre committed:
Update source file en.json

kogre committed:
New translations en.json (English)

kogre committed:
New translations en.yml (Luxembourgish)

kogre committed:
New translations en.json (Luxembourgish)

kogre committed:
New translations en.yml (Luxembourgish)

kogre committed:
New translations en.yml (Luxembourgish)

kogre committed:
New translations en.yml (Luxembourgish)

kogre committed:
New translations en.yml (Luxembourgish)

kogre committed:
Update source file en.json

kogre committed:
New translations en.json (German)

kogre committed:
New translations en.json (Luxembourgish)

kogre committed:
New translations en.json (English)

kogre committed:
Update source file en.json

kogre committed:
New translations en.json (Portuguese, Brazilian)

kogre committed:
New translations en.json (German)

IvaKop committed:
Swtich category with categories

brentguf committed:
Merge branch 'master' of https://github.com/CitizenLabDotCo/citizenlab into crowdin_master

brentguf committed:
Merge pull request #831 from CitizenLabDotCo/crowdin_master

New Crowdin updates

brentguf committed:
Up

brentguf committed:
Merge pull request #839 from CitizenLabDotCo/changelog/release-2021-10-06

changelog/release-2021-10-06

IvaKop committed:
Merge pull request #837 from CitizenLabDotCo/IN-568/replace-category-with-categories

IN-568 Replace deprecated "category" query parameter with "categories"

sergei-kucher commented:
@kogre I don't know what is going on and why github shows the wrong diff.
This link show the right diff between the branches
https://github.com/CitizenLabDotCo/citizenlab/compare/CL2-6613-customizable-navbar...navbar%23CL2-6698
